include/sparrow/layout/array_access.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "sparrow/arrow_array_schema_proxy.hpp"
+
+
+namespace sparrow::detail
+{
+    
+    class array_access
+    {
+    public:
+        template<class ARRAY>
+        static inline const sparrow::arrow_proxy& storage(const ARRAY& array) 
+        {
+            return array.storage();
+        }
+
+        template<class ARRAY>
+        static inline sparrow::arrow_proxy& storage(ARRAY& array)
+        {
+            return array.storage();
+        }
+
+        template<class ARRAY>
+        requires(std::is_rvalue_reference_v<ARRAY&&>)
+        static inline sparrow::arrow_proxy&& extract_arrow_proxy(ARRAY&& array)
```

Comment:
Same remark here (notice this is just about the return type, the implementation is correct).

Reply:
The return type should be `sparrow::arrow_proxy` instead.